src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests/ServiceControllerTests.cs

```diff
@@ -87,9 +87,16 @@ private static bool RunningWithElevatedPrivileges
             get { return s_runningWithElevatedPrivileges.Value; }
         }
 
-        private void AssertExpectedProperties(ServiceController testServiceController)
+        private void AssertExpectedProperties(ServiceController testServiceController, bool serviceNameToUpper = false)
         {
-            Assert.Equal(_testService.TestServiceName, testServiceController.ServiceName);
+            if (serviceNameToUpper)
```

Comment:
I didn't know Assert.Equal could take a comparer. If so it sounds reasonable. Will give it a try.